src/main/java/microsoft/exchange/webservices/data/core/EwsUtilities.java

```diff
@@ -729,16 +729,16 @@ public static String serializeEnum(Object value) {
         }
       }
       return null;
-    } else if (cls.isAssignableFrom(Double.class)) {
+    } else if (Double.class.isAssignableFrom(cls)) {
       return (T) ((Double) Double.parseDouble(value));
-    } else if (cls.isAssignableFrom(Number.class))  {
+    } else if (Number.class.isAssignableFrom(cls))  {
       return (T) ((Integer) Integer.parseInt(value));
```

Comment:
casting `Number` to `Integer` will fail the following tests:
```java
  @Test
  public void testParseBigInt() throws IllegalAccessException, ParseException, InstantiationException {
    Assert.assertEquals(BigInteger.TEN, EwsUtilities.parse(BigInteger.class, BigInteger.TEN.toString()));
  }

  @Test
  public void testParseBigDec() throws IllegalAccessException, ParseException, InstantiationException {
    Assert.assertEquals(BigDecimal.TEN, EwsUtilities.parse(BigDecimal.class, BigDecimal.TEN.toString()));
  }
```
maybe there should be a more general implementation here.